### PR TITLE
Use cluster metadata for destination_service_name

### PR DESCRIPTION
**Pull request by keithmattix**
**What this PR does / why we need it**: Fixes istio/istio#49842

**Special notes for your reviewer**:
~I'm setting the field to unknown right now just to confirm that we're retrieving the namespace from metadata correctly. Before we merge, it should probably fall back to peer metadata~ The only two tests that were failing were for clusters with no metadata; we're good to go here.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
LGTM with fallback to peer_namespace

**Comment by zirain** [comment]
LGTM, can you add a test for it?

**Comment by keithmattix** [comment]
The test is already kind of there; do you just want to see a diff between endpoint metadata and cluster metadata to prove the latter takes priority?

**Comment by zirain** [comment]
> The test is already kind of there; do you just want to see a diff between endpoint metadata and cluster metadata to prove the latter takes priority?

yeah, that's what I mean.

File: source/extensions/filters/http/istio_stats/istio_stats.cc [hunk 1]
```diff
@@ -1230,7 +1235,9 @@ class IstioStatsFilter : public Http::PassThroughFilter,
                                                                ? context_.unknown_
                                                                : pool_.add(service_host_name)});
       tags_.push_back({context_.destination_service_namespace_,
-                       !peer_namespace.empty() ? pool_.add(peer_namespace) : context_.unknown_});
+                       !service_namespace.empty()
+                           ? pool_.add(service_namespace)
+                           : (peer ? pool_.add(peer->namespace_name_) : context_.unknown_)});
```
Comment (zirain) [inline source/extensions/filters/http/istio_stats/istio_stats.cc:1240]: why change this from `!peer_namespace.empty() ? pool_.add(peer_namespace) : context_.unknown_`?

Comment (keithmattix) [reply source/extensions/filters/http/istio_stats/istio_stats.cc:1240]: We were relying exclusively on endpoint metadata for destination_service_namespace. This makes sense for workload based metrics, but for service metrics, it has the unfortunate side effect of losing information when there are no healthy backends. So instead, we petard with using cluster metadata if it exists, falling back to endpoint metadata after that

Comment (zirain) [reply source/extensions/filters/http/istio_stats/istio_stats.cc:1240]: just L1240, why not use prev expression?

Comment (keithmattix) [reply source/extensions/filters/http/istio_stats/istio_stats.cc:1240]: Ah I see what you mean; I'll switch it around so it's consistent

**Comment by keithmattix** [comment]
> > The test is already kind of there; do you just want to see a diff between endpoint metadata and cluster metadata to prove the latter takes priority?
> 
> yeah, that's what I mean.

Done!

**Review by zirain — APPROVED** [review]

**Pull request merged**